src/main/java/com/synopsys/integration/jenkins/scan/service/BlackDuckParametersService.java

```diff
@@ -0,0 +1,111 @@
+        return createBlackDuckObject(blackDuckParametersMap);
+    }
+
+    public Map<String, Object> prepareBlackDuckParameterValidation(Map<String, Object> blackDuckParametersFromPipeline) {
+        Map<String, Object> blackDuckParametersMapFromUI = createBlackDuckParametersMapFromJenkinsUI();
+        Map<String, Object> blackDuckParametersMap = getCombinedBlackDuckParameters(blackDuckParametersFromPipeline, blackDuckParametersMapFromUI);
+        return blackDuckParametersMap;
+    }
+
+    public BlackDuck createBlackDuckObject(Map<String, Object> blackDuckParametersMap) {
+        BlackDuck blackDuck = new BlackDuck();
+
+        for (Map.Entry<String, Object> entry : blackDuckParametersMap.entrySet()) {
+            String key = entry.getKey();
+            String value = entry.getValue().toString().trim();
+
+            switch (key) {
+                case ApplicationConstants.BLACKDUCK_URL_KEY:
+                    blackDuck.setUrl(value);
+                    break;
+                case ApplicationConstants.BLACKDUCK_API_TOKEN_KEY:
+                    blackDuck.setToken(value);
+                    break;
+                case ApplicationConstants.BLACKDUCK_INSTALL_DIRECTORY_KEY:
+                    blackDuck.setInstallDirectory(value);
+                    break;
+                case ApplicationConstants.BLACKDUCK_SCAN_FULL_KEY:
+                    if (value.equals("true") || value.equals("false")) {
+                        blackDuck.getScan().setFull(Boolean.parseBoolean(value));
+                    }
+                    break;
+                case ApplicationConstants.BLACKDUCK_SCAN_FAILURE_SEVERITIES_KEY:
+                    if (value.length() > 0) {
+                        List<String> failureSeverities = new ArrayList<>();
+                        String[] failureSeveritiesInput = value.toUpperCase().split(",");
+
+                        for (String input : failureSeveritiesInput) {
+                            failureSeverities.add(input.trim());
+                        }
+                        blackDuck.getScan().setFailureSeverities(failureSeverities);
+                    }
+                    break;
+                case ApplicationConstants.BLACKDUCK_AUTOMATION_FIXPR_KEY:
+                    if (value.equals("true") || value.equals("false")) {
+                        blackDuck.getAutomation().setFixpr(Boolean.parseBoolean(value));
+                    }
+                    break;
+                case ApplicationConstants.BLACKDUCK_AUTOMATION_PRCOMMENT_KEY:
+                    if (value.equals("true") || value.equals("false")) {
+                        blackDuck.getAutomation().setPrcomment(Boolean.parseBoolean(value));
+                    }
+                    break;
+            }
+        }
+
+        return blackDuck;
+    }
+
+    public boolean performParameterValidation(Map<String, Object> blackDuckParams) {
```

Comment:
If this method is targeting only blackduck parameter validation, we should rename it for blackduck specific.

Reply:
changed, Thanks!